### PR TITLE
Bug fix: Validate channel name according to pusher docs

### LINKED ISSUE #33: Validate channel name according to pusher docs
https://pusher.com/docs/client_api_guide/client_channels#naming-channels

### PR DESCRIPTION
**Pull request by burtonjc**
I have never done elixir before, so please let me know if I need additional test coverage anywhere.

closes #33

### CODE REVIEW

File: test/channel_test.exs [hunk 1]
```diff
@@ -48,4 +48,39 @@ defmodule Poxa.ChannelTest do
...
+    assert valid?("some-channel")
+    assert valid?("some=channel")
+    assert valid?("some@channel")
+    assert valid?("some,channel")
+    assert valid?("some.channel")
+    assert valid?("some;channel")
+    assert valid?("foo-bar_1234@=,.;")
+  end
+
+  test "valid? return false for valid characters" do
```
Comment (edgurgel) [inline test/channel_test.exs]: should this be "invalid" ?

**Comment by edgurgel** [comment]
Thanks for the PR :heart: ! The function had just 3 tests as part of the doctests and now we will have a good amount of tests :+1:.

I added just a comment but other than that it's good to go!

**Comment by burtonjc** [comment]
No problem, thanks for being so responsive! I just fixed that test name.

**Comment by edgurgel** [comment]
Thanks for the contribution! I will release a new minor version this week! :tada:

**Comment by burtonjc** [comment]
Great, thanks! :beers:

**Pull request merged**